The `7-base_geometry` module
======================

Using ``{function}``
-------------------

>>> BaseGeometry = __import__('7-base_geometry').BaseGeometry
    
>>> bg = BaseGeometry()

--- Testing area()  ---

>>> bg.area()
Traceback (most recent call last):
Exception: area() is not implemented

--- Base Case ---
>>> bg.integer_validator("Jon", 12)

--- Value not an int case ---
>>> bg.integer_validator("Jon", "Snow")
Traceback (most recent call last):
TypeError: Jon must be an integer

--- Value less than 0 ---
>>> bg.integer_validator("Jon", -1)
Traceback (most recent call last):
ValueError: Jon must be greater than 0

--- Value equal than 0 ---
>>> bg.integer_validator("Jon", 0)
Traceback (most recent call last):
ValueError: Jon must be greater than 0

--- name not a string ---
>>> bg.integer_validator(1, 2)

--- passing not the 2nd arg ---
>>> bg.integer_validator("Jon")
Traceback (most recent call last):
TypeError: integer_validator() missing 1 required positional argument: 'value'

--- Passing no args ---
>>> bg.integer_validator()
Traceback (most recent call last):
TypeError: integer_validator() missing 2 required positional arguments: 'name' and 'value'

--- Passing a list ---
>>> bg.integer_validator("Jon", [1, 2, 3])
Traceback (most recent call last):
TypeError: Jon must be an integer

--- Passing None as second arg ---
>>> bg.integer_validator("Jon", None)
Traceback (most recent call last):
TypeError: Jon must be an integer

--- Passing a tuple ---
>>> bg.integer_validator("Jon", (1, 2))
Traceback (most recent call last):
TypeError: Jon must be an integer

--- Passing an empty string ---
>>> bg.integer_validator("", 5)

--- Passing an empty string, but with exceptions ---
>>> bg.integer_validator("", "8")
Traceback (most recent call last):
TypeError:  must be an integer

--- Passing an empty string, but with exceptions ---
>>> bg.integer_validator("", 0)
Traceback (most recent call last):
ValueError:  must be greater than 0

--- Passing a bool ---
>>> bg.integer_validator("Jon", False)
Traceback (most recent call last):
TypeError: Jon must be an integer

--- Passing a set as second arg ---
>>> bg.integer_validator("Jon", {1, 2})
Traceback (most recent call last):
TypeError: Jon must be an integer